### ISSUE TITLE
Error ranges with unknown identifiers and multi-line strings with line continuation characters are incorrect

### ISSUE THREAD

**Issue by xperiandri** [dotnet/fsharp#9347]
I don't know how this kind of string writing is named so if this is a duplicate I'm sorry but I don't know how to search that issue.

![image](https://user-images.githubusercontent.com/2365592/83330309-ca3cd580-a296-11ea-94e0-2d120e786ff1.png)

**Repro steps**

Create a multi-line string with `\` at the end of each line and a literal that is concatenated into that string. Try to rename literal.

``` F#
let [<Literal>] UnionTag = "kind"

let query =
        "SELECT uzanto.id AS uzantoID, \
    	        ST_DISTANCE(biblioteko.location.geolocation, {'type': 'Point', 'coordinates': " + coordinatesParameter + "}) AS distance \
         FROM uzanto \
         JOIN biblioteko IN uzanto.bibliotekoj \
         WHERE EXISTS (\
    	     SELECT VALUE possession \
             FROM possession IN biblioteko.content \
             WHERE possession." + unionTag + " = 'Owned' \
               AND possession.registriID = " + registriIdParameter + ")"
```

**Expected behavior**

Literal is renamed correctly and an error displayed at a literal place
`
**Actual behavior**

An error displayed on the left of the correct location and renaming changes string left to the literal

**Related information**

* .NET Core 3.1.201, .NET Standard 2.1
* Visual Studio 2019 16.6 Enterprise ENG

**Comment by abelbraaksma** [comment]
I think it's called "string continuation character", and there are a lot of problems when using it, unfortunately, see #2729, #3380 and https://github.com/dotnet/fsharp/issues/7734 (showing errors when there are no errors, for instance).

But your situation isn't mentioned in the other bugs, so I don't think it's a duplicate.

**Comment by cartermp** [comment]
Thanks for filing @xperiandri and thanks for linking related issues with strings and line continuation chars @abelbraaksma  - I've renamed this to more accurate describe the issue. This is distinct from the other ones since it's ultimately about error reporting for unknown identifiers.

@dsyme this might be good to consider with string interoplation, as it may mean that situations like this will be more common.

**Comment by dsyme** [comment]
These bugs should be fixed, but I feel using both string continuation characters and multi-line single quote strings should both really be considered deprecated in favour of triple-quote strings.

**Comment by abelbraaksma** [comment]
@dsyme, I would agree, but we have nothing to replace them with yet. I mean, there's not simply way to remove the whitespace. The benefit of using `\` as line continuation is that the whitespace and the newline are stripped.

**Comment by dsyme** [comment]
This is a dup of #7734, and both are fixed by #10003, I'll keep #7734 as the canonical bug
